homeassistant/components/google_travel_time/__init__.py

```diff
@@ -51,9 +56,46 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
                     )
                     options[CONF_TIME] = None
         hass.config_entries.async_update_entry(config_entry, options=options, version=2)
-        _LOGGER.debug(
-            "Migration to version %s.%s successful",
-            config_entry.version,
-            config_entry.minor_version,
-        )
+    if config_entry.version == 2:
+        entity_registry = er.async_get(hass)
+        old_unique_id = config_entry.entry_id
+        new_unique_id = f"{config_entry.entry_id}_{ATTR_DURATION}"
+
+        if old_entity_id := entity_registry.async_get_entity_id(
+            "sensor", DOMAIN, old_unique_id
+        ):
+            new_entity_id = f"{old_entity_id}_{ATTR_DURATION}"
```

Comment:
We shouldn't change the entity id. That's considered user land and would for sure break automations.

Recently we did add a button in the frontend so someone could "reset" their entity id to what it would be today. This also takes into account translations which this migration does not.

Reply:
@joostlek Thinking about it.... Does this PR make sense at all if we don't change the entity_id and only the unique_id? When splitting up the attributes in entities we could just leave the "duration" sensor as the current sensor and only give the newly created sensors name and unique_id suffixes like "distance,origin".